pkg/virt-handler/node-labeller/cpu_plugin.go

```diff
@@ -0,0 +1,180 @@
+		for _, obsoleteModel := range obsoleteCPUsx86 {
+			if obsoleteModel == model {
+				isObsolete = true
+			}
+		}
+		if isObsolete {
+			continue
+		}
+
+		newFeatures := subtractMap(n.cpuInfo.models[model], basicFeaturesMap)
+		for feature := range newFeatures {
+			features[feature] = true
+		}
+
+		cpus = append(cpus, model)
+	}
+	return cpus, features
+}
+
+//loadHostCapabilities loads info about cpu models, which can host emulate
+func (n *NodeLabeller) loadHostCapabilities() error {
+	hostDomCapabilities, err := n.getDomCapabilities()
+	if err != nil {
+		return err
+	}
+
+	usableModels := make([]string, 0)
+	for _, mode := range hostDomCapabilities.CPU.Mode {
+		for _, model := range mode.Model {
+			if model.Usable == usableNo || model.Usable == "" {
+				continue
+			}
+			usableModels = append(usableModels, model.Name)
+		}
+	}
+
+	n.hostCapabilities.items = usableModels
+
+	return nil
+}
+
+//loadCPUInfo load info about all cpu models
+func (n *NodeLabeller) loadCPUInfo() error {
+	files, err := ioutil.ReadDir(nodeLabellerVolumePath + "cpu_map/")
+	if err != nil {
+		return err
+	}
+
+	models := make(map[string]cpuFeatures)
+	for _, f := range files {
+		fileName := f.Name()
+		if strings.HasPrefix(fileName, "x86_") {
+			features, err := loadFeatures(fileName)
+			if err != nil {
+				return err
+			}
+			cpuName := strings.TrimSuffix(strings.TrimPrefix(fileName, "x86_"), ".xml")
+
+			models[cpuName] = features
+		}
+	}
+
+	n.cpuInfo.models = models
+
+	return nil
+}
+
+func (n *NodeLabeller) getDomCapabilities() (HostDomCapabilities, error) {
+	domCapabilitiesFile := nodeLabellerVolumePath + "virsh_domcapabilities.xml"
+	hostDomCapabilities := HostDomCapabilities{}
+	err := getStructureFromXMLFile(domCapabilitiesFile, &hostDomCapabilities)
+
+	return hostDomCapabilities, err
+}
+
+//LoadFeatures loads features for given cpu name
+func loadFeatures(fileName string) (cpuFeatures, error) {
+	if fileName == "" {
+		return nil, fmt.Errorf("file name can't be empty")
+	}
+
+	cpuFeaturepath := getPathCPUFeatures(fileName)
+	features := FeatureModel{}
+	err := getStructureFromXMLFile(cpuFeaturepath, &features)
+	if err != nil {
+		return nil, err
+	}
+
+	modelFeatures := cpuFeatures{}
+	for _, f := range features.Model.Features {
+		modelFeatures[f.Name] = true
+	}
+	return modelFeatures, nil
+}
+
+//getPathCPUFeatures creates path where folder with cpu models is
+func getPathCPUFeatures(name string) string {
+	return nodeLabellerVolumePath + "cpu_map/" + name
+}
+
+func subtractMap(a, b map[string]bool) map[string]bool {
+	new := make(map[string]bool)
+	for k := range a {
+		if _, ok := b[k]; !ok {
+			new[k] = true
+		}
+	}
+	return new
+}
+
+//GetStructureFromXMLFile load data from xml file and unmarshals them into given structure
+//Given structure has to be pointer
+func getStructureFromXMLFile(path string, structure interface{}) error {
+	rawFile, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
```

Comment:
I wonder if we shouldn't log here on V(4) the rawFile? 
I'm sure it will be easier to understand what went wrong when debugging.
What do you think?